metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java

```diff
@@ -0,0 +1,258 @@
+
+    /**
+     * create hive connector partition service.
+     *
+     * @param metacatHiveClient    hive client
+     * @param hiveMetacatConverter metacat converter
+     * @param connectorConfig      connector config
+     * @return HiveConnectorPartitionService
+     */
+    @Bean
+    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "false")
+    public HiveConnectorPartitionService partitionService(
+        final IMetacatHiveClient metacatHiveClient,
+        final HiveConnectorInfoConverter hiveMetacatConverter,
+        final MetacatConnectorConfig connectorConfig
+    ) {
+        return new HiveConnectorPartitionService(
+            connectorConfig.getCatalogName(), metacatHiveClient, hiveMetacatConverter);
+    }
+
+    /**
+     * create hive connector fast partition service.
+     *
+     * @param metacatHiveClient    hive client
+     * @param hiveMetacatConverter metacat converter
+     * @param threadServiceManager thread service manager
+     * @param connectorConfig      connector config
+     * @param dataSource           data source
+     * @return HiveConnectorPartitionService
+     */
+    @Bean
+    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "true")
+    public HiveConnectorFastPartitionService fastHivePartitionService(
+        final IMetacatHiveClient metacatHiveClient,
+        final HiveConnectorInfoConverter hiveMetacatConverter,
+        final ThreadServiceManager threadServiceManager,
+        final MetacatConnectorConfig connectorConfig,
+        @Qualifier("hiveDataSource") final DataSource dataSource
+    ) {
+        return new HiveConnectorFastPartitionService(connectorConfig.getCatalogName(),
+            metacatHiveClient, hiveMetacatConverter,
+            connectorConfig, threadServiceManager,
+            dataSource);
+    }
+
+
+    /**
+     * create hive connector database service.
+     *
+     * @param metacatHiveClient    hive client
+     * @param hiveMetacatConverter metacat converter
+     * @param connectorConfig      connector config
+     * @return HiveConnectorDatabaseService
+     */
+    @Bean
+    public HiveConnectorDatabaseService hiveDatabaseService(
+        final IMetacatHiveClient metacatHiveClient,
+        final HiveConnectorInfoConverter hiveMetacatConverter,
+        final MetacatConnectorConfig connectorConfig) {
+        return new HiveConnectorDatabaseService(connectorConfig.getCatalogName(),
+            metacatHiveClient, hiveMetacatConverter);
+    }
+
+    /**
+     * create hive connector table service.
+     *
+     * @param metacatHiveClient            metacat hive client
+     * @param hiveMetacatConverters        hive metacat converters
+     * @param hiveConnectorDatabaseService hive database service
+     * @param metacatConnectorConfig       connector config
+     * @return HiveConnectorTableService
+     */
+    @Bean
+    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "false")
```

Comment:
This seems like you want this to be the default and then you want to override it when the property is true. In this case it might be cleaner to say `@ConditionalOnMissingBean(HiveConnectorTableService.class)`. This way it's a catch all for any other case we might add later for implementation of these services. Example: Lets say one day we have a HiveConnectorTableService implementation that's not fast and not the default but then we start enabling properties for it. Well now this condition needs to become a compound conditional and each time we add a new one a new condition. It doesn't scale as cleanly as just being a catch all.